pkg/sql/compile/compile.go

```diff
@@ -1112,16 +1140,33 @@ func (c *Compile) fillAnalyzeInfo() {
 
 func (c *Compile) generateNodes(n *plan.Node) (engine.Nodes, error) {
 	var err error
+	var db engine.Database
+	var rel engine.Relation
 	var ranges [][]byte
 	var nodes engine.Nodes
 
-	db, err := c.e.Database(c.ctx, n.ObjRef.SchemaName, c.proc.TxnOperator)
+	db, err = c.e.Database(c.ctx, n.ObjRef.SchemaName, c.proc.TxnOperator)
 	if err != nil {
 		return nil, err
 	}
-	rel, err := db.Relation(c.ctx, n.TableDef.Name)
+	rel, err = db.Relation(c.ctx, n.TableDef.Name)
 	if err != nil {
-		return nil, err
+		var e error // avoid contamination of error messages
+		db, e = c.e.Database(c.ctx, "temp-db", c.proc.TxnOperator)
+		if e != nil {
+			return nil, e
+		}
+		rel, e = db.Relation(c.ctx, n.ObjRef.SchemaName+"-"+n.TableDef.Name)
```

Comment:
This seems to be a very bad idea.    How can you tell the different between schema   a, table b-c, and schema a-b, table c?

Reply:
will add test later

Reply:
This is referenced to the initMemoryEngineNoDist function.